compiler/src/dotty/tools/dotc/quoted/PickledQuotes.scala

```diff
@@ -125,6 +125,18 @@ object PickledQuotes {
               val quotedType = evalHole.nn.apply(idx, reifyTypeHoleArgs(args))
               PickledQuotes.quotedTypeToTree(quotedType)
           }
+        case tree @ Template(constr, parents, self, _) if tree.derived.isEmpty =>
```

Comment:
Can tree.derived not be empty here?

Reply:
I copied it from the `TreeMap` case. Not sure why it is there. It was introduced in fb09e828e2a9cd1320d036e3916555b08fbde52b.

Reply:
Following up in  #18368